src/backend/libc/io/syscalls.rs

```diff
@@ -12,6 +12,8 @@ use super::super::offset::{libc_preadv, libc_pwritev};
 #[cfg(all(target_os = "linux", target_env = "gnu"))]
 use super::super::offset::{libc_preadv2, libc_pwritev2};
 use crate::fd::{AsFd, BorrowedFd, OwnedFd, RawFd};
+#[cfg(any(target_os = "solaris", target_os = "illumos"))]
```

Comment:
To make the cfg's slightly easier to manage, we keep the contents of any/all alphabetized, so illumos should come before solaris.

Another option would be to do a test in build.rs and do `println!("cargo:rustc-cfg=solarish");` so that the code in rustix can just be `cfg(solarish)`.